pkg/lib/cockpit.d.ts

```diff
@@ -368,4 +368,134 @@ declare module 'cockpit' {
 
     export let localStorage: Storage;
     export let sessionStorage: Storage;
+
+    /**
+     * Options passed to the HTTP client constructor.
+     */
+    export interface HttpOptions {
+        /**
+         * The target address. If omitted, the endpoint string must include the host.
+         */
+        address?: string;
+        /**
+         * The port to connect to.
+         */
+        port?: number;
+        /**
+         * Enable TLS (HTTPS) connection.
+         */
+        tls?: boolean;
+        /**
+         * Whether to connect as superuser.
+         */
+        superuser?: boolean;
+        /**
+         * If true, responses will be delivered as binary (Uint8Array) instead of string.
+         */
+        binary?: boolean;
+        /**
+         * Default HTTP headers to send with every request.
+         */
+        headers?: { [key: string]: string };
+        /**
+         * Default query parameters to include with every request.
+         */
+        params?: { [key: string]: string | number };
+    }
+
+    /**
+     * Allowed types for HTTP request bodies.
+     * A body may be a string or a plain JavaScript object that will be encoded as JSON.
+     * If the body is undefined or null, an empty HTTP body is sent.
+     */
+    export type HttpBody = string | { [key: string]: any } | null | undefined;
+
+    /**
+     * Options for a single HTTP request.
+     */
+    export interface HttpRequestOptions {
+        /**
+         * The path (or endpoint) of the request.
+         */
+        path: string;
+        /**
+         * The HTTP method to use, e.g. "GET", "POST", etc.
+         */
+        method: string;
+        /**
+         * Headers specific to this request.
+         */
+        headers?: { [key: string]: string };
+        /**
+         * Query parameters specific to this request.
+         */
+        params?: { [key: string]: string | number };
+        /**
+         * Request body. Can be a string or a plain object to be JSON encoded.
+         */
+        body?: HttpBody;
```

Comment:
Curiously that's not true. I read pkg/lib/cockpit.js and the only supported values are `string | Uint8Array`. This "plain object" jsonification is only implemented in the `.post()` method specifically. Sorry for the mess -- as I said, ancient API worth being rewritten :cry: